tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1152,6 +1153,48 @@ def test_external_new_user_login_and_check_count_rhsso(
         assert error.typename == "NavigationTriesExceeded"
 
 
+@pytest.mark.skip_if_open("BZ:1873439")
+@destructive
+def test_login_failure_rhsso_user_if_internal_user_exist(
+    enable_external_auth_rhsso, rhsso_setting_setup, session, module_org, module_loc
+):
+    """Verify the failure of login for the external rhsso user in case same username
+    internal user exists
+
+    :id: e573902c-ed1a-11ea-835a-d46d6dd3b5b2
+
+    :BZ: 1873439
+
+    :CaseImportance: High
+
+    :steps:
+        1. create an internal user
+        2. create a rhsso user with same username mentioned in internal user
+        3. update the satellite to use rhsso and now try login using external rhsso user
+
+    :expectedresults: external rhsso user should not able to login with same username as internal
+    """
+    client_id = get_rhsso_client_id()
+    username = gen_string('alpha')
+    entities.User(
+        admin=True,
+        default_organization=module_org,
+        default_location=module_loc,
+        login=username,
+        password=settings.rhsso.password,
+    ).create()
+    external_rhsso_user = create_new_rhsso_user(client_id, username=username)
+    login_details = {
+        'username': external_rhsso_user['username'],
+        'password': settings.rhsso.password,
+    }
+    with Session(login=False) as rhsso_session:
+        with raises(NavigationTriesExceeded) as error:
+            rhsso_session.rhsso_login.login(login_details)
+            rhsso_session.task.read_all()
```

Comment:
I feel this part would look much nicer if written somewhat like:

```python
rhsso_session.rhsso_login.login(login_details)
login_page = rhsso_session.rhsso_login.read()
assert login_page['error_message']
assert "I'm not letting you in!" in login_page['error_message']
assert not login_page['logged_in']
# etc., you get the idea
```

So - try to login, call `read` on login page and run some explicit assertions, instead of asserting by-product of failed login, which is exception raised by navmazing.

Since that would require change in Airgun first, I don't consider it blocking for PR itself. But something to think about.

Reply:
@mirekdlugosz what level of effort would be needed to make that change in airgun?

Reply:
@mirekdlugosz I thought about that and will required to wait till upstream PR get merged, due to importance of this BZ I just wanted to add it So I dont miss this and anyways this will be skip till and when the BZ get resolved we could see this test is up and running. 

Anyone can ask me why to do this double work ? As I already have all the sauces/required helper functions, this writing test did not took not more than 20 mins hence I did it. 

@mirekdlugosz As this is unblocking I will add airgun issue linkinf with this PR so we will not miss your suggestion.

Reply:
@JacobCallahan it's impossible to give accurate estimate without looking at pages themselves, but I imagine it should break down to adding couple of widgets to view and new `read()` method to entity. And thinking whether it makes more sense to call `read()` explicitly, or at the end of `login()` method, or both.

The bigger problem is, Omkar is only other active reviewer in Airgun, so if he raises PR there, we will probably wait weeks to have it merged. That's why I don't consider this suggestion blocking.

@omkarkhatavkar adding ticket to Airgun so we can get back to this later works for me. Thanks.

Reply:
raised here https://github.com/SatelliteQE/airgun/issues/513